packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,566 @@
+      blobs: [],
+    }
+    try {
+      const gatewayStatusResponse: GatewayStatusResponse = await this.gatewayApiClient.status.getCurrent()
+      const notarizedTransaction = await TransactionBuilder.new().then((builder) =>
+        builder
+          .header({
+            networkId: networkId,
+            startEpochInclusive: gatewayStatusResponse.ledger_state.epoch,
+            endEpochExclusive: gatewayStatusResponse.ledger_state.epoch + 10,
+            nonce: generateRandomNonce(),
+            notaryPublicKey: radixPrivateKey.publicKey(),
+            notaryIsSignatory: true,
+            tipPercentage: 0,
+          })
+          .manifest(manifest)
+          .notarize(radixPrivateKey),
+      )
+
+      const compiledTransaction = await RadixEngineToolkit.NotarizedTransaction.compile(notarizedTransaction)
+      const compiledTransactionHex = Convert.Uint8Array.toHexString(compiledTransaction)
+      await this.broadcastTx(compiledTransactionHex)
+      const transactionId = await RadixEngineToolkit.NotarizedTransaction.intentHash(notarizedTransaction)
+      return transactionId.id
+    } catch (error) {
+      throw new Error('Failed to transfer')
+    }
+  }
+  /**
+   * Submits a transaction
+   * @param txHex - The transaction hex build with the transfer method
+   * @returns - The response from the gateway
+   */
+  async broadcastTx(txHex: string): Promise<string> {
+    const transactionSubmitResponse: TransactionSubmitResponse =
+      await this.gatewayApiClient.transaction.innerClient.transactionSubmit({
+        transactionSubmitRequest: {
+          notarized_transaction_hex: txHex,
+        },
+      })
+    return JSON.stringify(transactionSubmitResponse)
+  }
+
+  /**
+   * Prepares a transaction to be used by the transfer method
+   * It will include a non signed transaction
+   * @param params - The transaction params
+   * @returns a PreparedTx
+   */
+  async prepareTx(params: TxParams): Promise<PreparedTx> {
+    if (params.asset == undefined) {
+      throw new Error('asset can not be undefined')
+    }
+    const fromAddress = await this.getAddressAsync()
+    const stringManifest = `
+    CALL_METHOD
+      Address("${fromAddress}")
+      "lock_fee"
+      Decimal("${this.feeBounds.upper}");
+    CALL_METHOD
+      Address("${fromAddress}")
+      "withdraw"
+      Address("${params.asset.symbol}")
+      Decimal("${params.amount.amount()}");
+    TAKE_FROM_WORKTOP
+      Address("${params.asset.symbol}")
+      Decimal("${params.amount.amount()}")
+      Bucket("xrd_payment");
+    CALL_METHOD
+      Address("${params.recipient}")
+      "try_deposit_or_abort"
+      Bucket("xrd_payment")
+      None;
+    `
+    const preparedTx = {
+      rawUnsignedTx: stringManifest,
+    }
+    return preparedTx
+  }
+
+  /**
+   * Get asset information.
+   * @returns Asset information.
+   */
+  getAssetInfo(): AssetInfo {
+    const assetInfo: AssetInfo = {
```

Comment:
Minor: Perhaps just directly return instead of assining to a variable first?

Reply:
done